### PR TITLE
chore(internal): fix CI Visibility usage of the core API

### PR DESCRIPTION
**Pull request by brettlangdon**
Fixes some small inconsistencies with the core api usage in CI Visibility.

1. A call to `dispatch_with_results` when no value is expected to be returned (small performance optimization to avoid building a results dictionary when it isn't needed).
2. A call to `dispatch_with_results` which would not return the underlying listener value
    a. (I _think_ it would always return a falsey value since it should have been returning the `core._MissingEvent` `EventResult` which is falsey and which has a value of `None`... ?)

I am not sure which tests should be evaluated/updated, since I am surprised there weren't any failing tests before, there _should_ have been.

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by emmettbutler — APPROVED** [review]
Oh yeah, it's not clear at all to me why this isn't causing `test_is_item_itr_skippable_test_level` to fail in current main. This change does look more correct, though.

File: ddtrace/ext/ci_visibility/api.py [hunk 1]
```diff
@@ -239,7 +239,7 @@ def finish(
     @_catch_and_log_exceptions
     def get_settings(item_id: Optional[CISessionId] = None):
         log.debug("Getting settings for session %s", item_id)
-        core.dispatch_with_results("ci_visibility.session.get_settings", (item_id,))
+        core.dispatch("ci_visibility.session.get_settings", (item_id,))
```
Comment (romainkomorndatadog) [inline ddtrace/ext/ci_visibility/api.py:242]: The actual fix here is to actually return the settings.

**Review by romainkomorndatadog — CHANGES_REQUESTED** [review]
These (the settings request) aren't in use yet so they're not tested (and, clearly, not working either).

edit: the external API `is_item_itr_skippable()` static method (that ends up calling `core.dispatch_with_results("ci_visibility.itr.is_item_skippable")` itself) is also not tested yet.

`test_is_item_itr_skippable_test_level` is directly testing the `CIVisibility.is_item_itr_skippable()` function directly (which is why it's in `test_ci_visibility.py`), so any changes in this PR wouldn't affect its results.

I'll take over the PR.

File: ddtrace/ext/ci_visibility/api.py (hunk 1 above)
Comment (romainkomorndatadog) [reply ddtrace/ext/ci_visibility/api.py:242]: Actually, since I don't have a use case for this at the moment, I'm just going to delete it.

**Comment by romainkomorndatadog** [comment]
Renaming this to a `chore` because this API isn't released yet, so it's not a "fix".

**Review by romainkomorndatadog — APPROVED** [review]

**Review by emmettbutler — APPROVED** [review]

**Pull request merged**